img_proof/ipa_utils.py

```diff
@@ -131,7 +131,7 @@ def expand_test_files(test_dirs, names, exclude):
             ' and/or test descriptions.'
         )
 
-    if not isinstance(exclude, list):
+    if not isinstance(exclude, tuple):
```

Comment:
This will cause a regression when no input is provided for exclude since the default is an empty list. In this we don't care if the value is a list, set, tuple etc but we want to prevent an accidental string input because this leads to very confusing errors. Thus I think it's best that we check if it's a string in the guard clause instead.

```
if isinstance(exclude, str):
        raise IpaUtilsException(
            'Exclude should be a list, tuple or set containing test names.'
        )
```

Reply:
Updated, please check again

Reply:
Well that fixes the regression. However, if we allow list and tuple it makes sense to allow anything that's iterable. But str will act like an interable in a loop thus it makes sense to use a guard clause only on str.img-proof can be used as an SDK and thus it's likely a user would provide a set of excludes which seems a perfectly valid use case.

Reply:
We could check for an iterable type and in the case of a single string, convert it to a list. That would fulfil your requirement plus allows also for a single string argument.